lib/uuidUtils.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+const uuidRegEx = /^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$/;
+const uuidV1Offset = 122192928000000000; //  100 nanosecond intervals since 15 October 1582
+
+const uuidUtils = {};
+
+/**
+ * Extract time in nanoseconds from UUID V1
+ *
+ * @param {string} uuidv1
+ * @return {Integer|null}
+ *
+ */
+uuidUtils.getTime = (uuidv1) => {
+    const uuidComponents = uuidv1.split('-');
+
+    return parseInt(
+        [
+            uuidComponents[2].substring(1),
+            uuidComponents[1],
+            uuidComponents[0]
+        ].join(''), 16);
+};
+
+/**
+ * Extract time in nanoseconds from UUID V1
+ *
+ * @param {string} uuidv1
+ * @return {Date|null}
+ *
+ */
+uuidUtils.getDate = (uuidv1) => {
+    if (!(typeof uuidv1 === 'string' || uuidv1 instanceof String)) {
```

Comment:
Move this check to `getTime()` since this function uses that one. Also, better to return `undefined` rather than `null`

Reply:
Changed to undefined